File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/PolarOfflineRecordingTriggerMode.kt
package com.polar.sdk.api.model

import com.polar.sdk.api.PolarBleApi

/**
 * Polar offline recording trigger mode. Based on the trigger mode the
 * device starts the offline recording automatically
 */
enum class PolarOfflineRecordingTriggerMode {
    // The automatic start of offline recording is disabled
    TRIGGER_DISABLED,

    // Triggers the offline recording when device is powered on
    TRIGGER_SYSTEM_START,

    // Triggers the offline recording when exercise is started in device
    TRIGGER_EXERCISE_START
}

/**
 * Polar offline recording trigger
 *  @property triggerMode the mode of the trigger
 *  @property triggerFeatures features enabled with the trigger, empty if [triggerMode] is [PolarOfflineRecordingTriggerMode.TRIGGER_DISABLED].
 *  In case of the [PolarBleApi.PolarDeviceDataType.PPI] and [PolarBleApi.PolarDeviceDataType.HR] the [PolarSensorSetting] is null
 */
data class PolarOfflineRecordingTrigger(
    val triggerMode: PolarOfflineRecordingTriggerMode,
    val triggerFeatures: Map<PolarBleApi.PolarDeviceDataType, PolarSensorSetting?>
)

File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/PolarOfflineRecordingTriggerMode.swift
//  Copyright © 2023 Polar. All rights reserved.

import Foundation

/// Polar offline recording trigger mode. Offline recording trigger can be used to start the offline recording automatically in device, based on selected trigger mode.
public enum PolarOfflineRecordingTriggerMode {
    // The automatic start of offline recording is disabled
    case triggerDisabled
    
    // Triggers the offline recording when device is powered on
    case triggerSystemStart
    
    // Triggers the offline recording when exercise is started in device
    case triggerExerciseStart
}

/// Polar offline recording trigger
public struct PolarOfflineRecordingTrigger {
    /// The mode of the trigger
    public let triggerMode: PolarOfflineRecordingTriggerMode
    
    /// Dictionary containing the `PolarDeviceDataType` keys for enabled triggers. Dictionary
    /// is empty if `triggerMode` is `PolarOfflineRecordingTriggerMode.triggerDisabled`.
    /// In case of the `PolarDeviceDataType.ppi` or `PolarDeviceDataType.hr`  the `settings` is nil
    public let triggerFeatures: [PolarDeviceDataType : PolarSensorSetting?]
    
    public init(triggerMode: PolarOfflineRecordingTriggerMode, triggerFeatures: [PolarDeviceDataType : PolarSensorSetting?] ) {
        self.triggerMode = triggerMode
        self.triggerFeatures = triggerFeatures
    }
}
